GetStream/stream-chat-react-native
# fix: avoid requesting app settings if the client isn't connected

## 🎯 Goal

Currently, if you render a Chat component with a client that hasn't yet been connected, you'll get the following error:

```
An error occurred while getting app settings: Error: Both secret and user tokens are not set. Either client.connectUser wasn't called or client.disconnect was called
```

This fixes #1634

## 🛠 Implementation details

In addition to checking whether the app is online before requesting app settings, I've added a check for `client.userID`, which will be set once `client.connectUser` has been called.

## 🎨 UI Changes

None

## 🧪 Testing

1. Create an app with a Chat component without any children
2. Add a client to the Chat component, without calling `client.connectUser` first
3. See that Metro doesn't write any error to the console

## ☑️ Checklist
